software/teachee-desktop/src/app.rs

```diff
@@ -187,8 +194,59 @@ impl eframe::App for App {
                 ui.separator();
                 ui.menu_button("File", |ui| {
                     if ui.button("Export to CSV").clicked() {
-                        todo!("Exporting to CSV");
+                        let epoch_time = SystemTime::now()
+                            .duration_since(UNIX_EPOCH)
+                            .unwrap()
+                            .as_secs();
+                        let pid = std::process::id();
+                        let file_name = format!("{}-{}-{}", epoch_time, pid, "teachee.csv");
+                        let file = match File::create(file_name.clone()) {
+                            Ok(a) => a,
+                            Err(_) => {
+                                let err_msg = format!("Unable to open file: {}", file_name);
+                                thread::spawn(move || {
+                                    MessageDialog::new()
+                                        .set_text(&err_msg)
+                                        .show_alert()
+                                        .unwrap();
+                                });
+                                return;
+                            }
+                        };
+                        let mut writer = Writer::from_writer(file);
+                        let (condvar, mutex) = &*data.bufs[*buf_idx];
+                        let mut buf_state = condvar
+                            .wait_while(mutex.lock().unwrap(), |buf_state| buf_state.is_empty())
+                            .unwrap();
+                        let (channels, num_samples) = buf_state.unwrap();
+
+                        let v_strs = channels.voltage1[0..num_samples]
+                            .iter()
+                            .map(|e| e.to_string());
+
+                        let do_writing = move || -> Result<(), Box<dyn Error>> {
+                            writer.write_field("current1")?;
+                            writer.write_record(v_strs)?;
+                            let c_strs = channels.current1[0..num_samples]
```

Comment:
Move this to the same place as v_strs
Also we should name it "Channel 1" and "Channel 2" being voltage1 and current1 arrays since that's what the user sees

Reply:
d37164cb785ec711aa6672fad084e656b47224da

Reply:
Is there a reason c_strs has to be created in do_writing and v_strs outside? I say it makes more sense to create v_strs and c_strs both before since do_writing should only be concerned with writing with `writer`.